src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -73,31 +82,154 @@ internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol origina
 
             var numElements = elementTypes.Length;
 
-            // PROTOTYPE handling for greater than 8
-            if (numElements <= 1 || numElements > 7)
+            if (numElements <= 1)
             {
                 throw ExceptionUtilities.Unreachable;
             }
+            var underlyingTypeChain = GetTupleUnderlyingTypeChain(elementTypes, syntax, binder, diagnostics);
+
+            var fieldsBuilder = ArrayBuilder<TupleFieldSymbol>.GetInstance(numElements);
+
+            var hasChunks = numElements > 7;
+            if (!hasChunks)
+            {
+                // short tuple: each member gets a simple field symbol
+                for (int elementIndex = 0; elementIndex < numElements; elementIndex++)
+                {
+                    var wellKnownTypeMember = GetTupleTypeMember(numElements, elementIndex);
+                    var underlyingField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, wellKnownTypeMember, diagnostics, syntax: syntax);
+
+                    var field = new TupleFieldSymbol(elementNames.IsDefault ? TupleMemberName(elementIndex) : elementNames[elementIndex],
+                                               this,
+                                               elementTypes[elementIndex],
+                                               OneOrMany.Create(underlyingField?.AsMember(underlyingTypeChain[0])));
+                    fieldsBuilder.Add(field);
+                }
```

Comment:
It is, and I would like to merge the two cases, but I'm trying to avoid creating an array of one, since short tuples are the common case.